zerver/lib/event_schema.py

```diff
@@ -54,3 +54,19 @@ def check_events_dict(
         check_string,
     ])),
 ])
+
+update_display_settings_schema = check_events_dict(
+    required_keys=[
+        ('type', equals('update_display_settings')),
+        ('setting_name', check_string),
+        ('user', check_string),
+        ('setting', check_union([
+            check_bool,
+            check_int,
+            check_string,
+        ])),
```

Comment:
It's a little sad to me that we're migrating away from something with a consistent type for `setting` to something without that.  

Is the idea that `assert isinstance(event['setting'], property_type)` effectively does this check for us?  

I feel like I want that to be a bit more obvious somehow; as it is I feel like someone writing a new test based on the existing test has a good chance of forgetting that critical assert.

Reply:
I think my change here just clarifies the intrinsic problem, which is that our event's fundamental type is a union type.

I wouldn't get overly hung up on `test_events` missing schema checks.  It's number one job is to make sure that the re-apply-event logic works.  The new paradigm here will be that we use the schema check for basic sanity checks and then test specific data elements more closely.

Reply:
We actually want `test_events` to do more than it does now in terms of data validation.  The schemas were partly just a lazy way to get around the fact that some things like database ids are non-deterministic, so we just punted and made sure the types of the data were valid.

Reply:
> 
> Is the idea that `assert isinstance(event['setting'], property_type)` effectively does this check for us?

yep
 
> I feel like I want that to be a bit more obvious somehow; as it is I feel like someone writing a new test based on the existing test has a good chance of forgetting that critical assert.

I think we can catch this in code review.  We don't add a ton of new events to our system in any given month, and most coding errors are likely to be failure to write the `apply_events` code correctly, which usually involves some senior programmer intervention.